build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoRun.java

```diff
@@ -0,0 +1,218 @@
+     */
+    public GoRun(String goCmdArgs, Path path, ArtifactoryBuildInfoClientBuilder clientBuilder, String repo, String username, String password, Log logger, Map<String, String> env) throws IOException {
+        super(clientBuilder, path, logger);
+        this.goCmdArgs = goCmdArgs;
+        this.resolutionRepository = repo;
+        this.resolverUsername = username;
+        this.resolverPassword = password;
+        this.env = env;
+    }
+
+    public Build execute() {
+        try (ArtifactoryBuildInfoClient artifactoryClient = (clientBuilder != null ? (ArtifactoryBuildInfoClient) clientBuilder.build() : null)) {
+            if (artifactoryClient != null) {
+                preparePrerequisites(resolutionRepository, artifactoryClient);
+                setResolverAsGoProxy(artifactoryClient);
+            }
+            this.goCommandExecutor = new CommandExecutor(GO_CLIENT_CMD, env);
+            runCmd();
+            collectDependencies();
+            return createBuild(null, dependenciesList);
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    /*
+     * In order to use Artifactory as a resolver we need to set GOPROXY env var with Artifactory details.
+     * Wa also support fallback to VCS in case pkg doesn't exist in Artifactort,
+     */
+    private void setResolverAsGoProxy(ArtifactoryBuildInfoClient client) throws Exception {
+        URL rtUrl = new URL(client.getArtifactoryUrl());
+        URIBuilder proxyUrlBuilder = new URIBuilder()
+                .setScheme(rtUrl.getProtocol())
+                .setUserInfo(resolverUsername, resolverPassword)
+                .setHost(rtUrl.getHost())
+                .setPath(rtUrl.getPath() + ARTIFACTORY_GO_API + resolutionRepository);
+        String proxyValue = proxyUrlBuilder.build().toURL().toString() + "," + GOPROXY_VCS_FALLBACK;
+        env.put(GOPROXY_ENV_VAR, proxyValue);
+    }
+
+    private void runCmd() throws Exception {
+        List<String> args = new ArrayList<>();
+        for (String arg : goCmdArgs.split(" ")) {
+            args.add(arg);
+        }
+        runGoCmd(args);
+    }
+
+    /*  Run 'go mod graph' and parse its output.
+     *  This command might change go.mod and go.sum content, so we backup and restore those files.
+     *    The output format is:
+     *     * For direct dependencies:
+     *          <module-name> <dependency's-module-name>@v<dependency-module-version>
+     *     * For transient dependencies:
+     *        <dependency's-module-name>@v<dependency-module-version> <dependency's-module-name>@v<dependency-module-version>
+     *   In order to populate build info dependencies, we parse the second column uf the mod graph output.
+     */
+    private void collectDependencies() throws Exception {
+        backupModAnsSumFiles();
+        List<String> goModArgs = new ArrayList<>();
```

Comment:
```suggestion
        List<String> goModArgs = Lists.newArrayList(GO_MOD_GRAPH_CMD.split(" "));
```

Reply:
Same as before - Can we avoid using guava?